File: shared/src/commonMain/kotlin/com/medistock/shared/config/DebugConfig.kt
package com.medistock.shared.config

/**
 * Debug configuration for the MediStock SDK.
 * Controls debug logging and other development features.
 */
object DebugConfig {
    /**
     * Enable or disable debug logging throughout the application.
     * Set to false for production builds.
     */
    var isDebugEnabled: Boolean = false
        private set

    /**
     * Enable debug mode.
     * Should only be called during app initialization in debug builds.
     */
    fun enableDebug() {
        isDebugEnabled = true
    }

    /**
     * Disable debug mode.
     * Should be called for production builds.
     */
    fun disableDebug() {
        isDebugEnabled = false
    }

    /**
     * Log a debug message if debug mode is enabled.
     */
    inline fun log(tag: String, message: () -> String) {
        if (isDebugEnabled) {
            println("[$tag] ${message()}")
        }
    }

    /**
     * Log a debug message if debug mode is enabled.
     */
    fun log(tag: String, message: String) {
        if (isDebugEnabled) {
            println("[$tag] $message")
        }
    }
}

File: iosApp/iosApp/Services/DebugConfig.swift
import Foundation

/// Debug configuration for the MediStock iOS app.
/// Controls debug logging and other development features.
final class DebugConfig {
    static let shared = DebugConfig()

    private static let debugModeKey = "debug_mode"

    /// Enable or disable debug logging throughout the application.
    /// Persisted in UserDefaults, can be toggled from admin settings.
    private(set) var isDebugEnabled: Bool = false

    private init() {
        if UserDefaults.standard.object(forKey: Self.debugModeKey) != nil {
            isDebugEnabled = UserDefaults.standard.bool(forKey: Self.debugModeKey)
        } else {
            #if DEBUG
            isDebugEnabled = true
            #endif
        }
    }

    /// Enable debug mode manually and persist the setting.
    func enableDebug() {
        isDebugEnabled = true
        UserDefaults.standard.set(true, forKey: Self.debugModeKey)
    }

    /// Disable debug mode manually and persist the setting.
    func disableDebug() {
        isDebugEnabled = false
        UserDefaults.standard.set(false, forKey: Self.debugModeKey)
    }

    /// Log a debug message if debug mode is enabled.
    /// - Parameters:
    ///   - tag: The tag/category for the log message
    ///   - message: The message to log (autoclosure for lazy evaluation)
    func log(_ tag: String, _ message: @autoclosure () -> String) {
        guard isDebugEnabled else { return }
        print("[\(tag)] \(message())")
    }

    /// Log a debug message if debug mode is enabled.
    /// - Parameters:
    ///   - tag: The tag/category for the log message
    ///   - message: The message to log
    func log(_ tag: String, message: String) {
        guard isDebugEnabled else { return }
        print("[\(tag)] \(message)")
    }
}

/// Global debug logging function for convenience
/// - Parameters:
///   - tag: The tag/category for the log message
///   - message: The message to log
func debugLog(_ tag: String, _ message: @autoclosure () -> String) {
    DebugConfig.shared.log(tag, message())
}
